### ISSUE TITLE
Invoke context doesn't work for Group.run

### ISSUE THREAD

**Issue by rahul003** [fabric/fabric#1861]
```
c = Context()
with c.cd(dir):
    print(c.cwd)
    self.group.run('pwd', **kwargs, hide=hide)
```

c.cwd shows correct directory but group.run doesn't show that path. How do I get around this?

**Comment by stevenhair** [comment]
I don't think that it's possible to use contexts with groups right now. For the moment, my workaround is to just loop through my hosts and create a connection for each one.

Naturally, this isn't the best solution. It's a bit messy, verbose, and doesn't really work if you need a `ThreadingGroup`.

```python
for host in ['host1', 'host2', 'host3]:
    with Connection(host) as c:
        with c.cd('/some/dir'):
            c.run('pwd')

```

**Comment by Eidansoft** [comment]
I face this issue, Contexts not working at Groups. The work-around I used is concatenate the `cd` with my command inside a `/bin/bash -c ...` line (in order to make the special chars like `&&` been not escaped from the `sudo`):
`
conn_group.sudo(f'/bin/bash -c "cd {working_path} && find {working_path} -name \'*.pyc\' -delete"', user=user)
`
